### PR TITLE
Exclude training support from BatchNorm in minimal build

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Previous changes [excluded](https://github.com/microsoft/onnxruntime/pull/7703) training specific code but that was recently [undone](https://github.com/microsoft/onnxruntime/pull/8269) to support a pytorch CI need that isn't relevant to minimal builds.

**Motivation and Context**
Reduce binary size in minimal build.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/nn/batch_norm.h [hunk 1]
```diff
@@ -28,28 +28,40 @@
...
-                  "Output running_mean and running_var are valid and required for training mode.");
     } else {
       is_train_ = OpKernel::Node().OutputDefs().size() > 1;
     }
 
-    ORT_ENFORCE(!is_train_ || is_spatial_, "Training mode does not support non-spatial BN");
+    if (is_train_) {
+#if defined(BATCHNORM_INCLUDE_TRAINING_SUPPORT)
+      size_t output_count = op_kernel_info.node().OutputDefs().size();
+      ORT_ENFORCE((is_train_ && output_count == 3) || (!is_train_ && output_count == 1),
```
Comment (gwang-msft) [inline onnxruntime/core/providers/cpu/nn/batch_norm.h:55]: Move this outside of ` if (is_train_)`?
And isn't `output_count == 3` opset 14 only?

Comment (skottmckay) [reply onnxruntime/core/providers/cpu/nn/batch_norm.h:55]: You're right - I overlooked it changing from 5 to 3. The ONNX type/shape inferencing checks the number of outputs is valid for opset 14 though so we don't need to do that again here.

Based on that I don't think the ORT_ENFORCE is needed as we didn't have it prior to opset 14.

**Review by gwang-msft — APPROVED** [review]

**Review by SherlockNoMad — APPROVED** [review]
